dds/idl/marshal_generator.cpp

```diff
@@ -405,6 +438,19 @@ namespace {
 
     AST_Type* elem = resolveActualType(seq->base_type());
     Classification elem_cls = classify(elem);
+
+    AST_Decl* node = td;
+    if (td == 0) {
+      node = seq;
+    }
+    const ExtensibilityKind exten = be_global->extensibility(node);
+    const OpenDDS::DataRepresentation repr =
+      be_global->data_representations(node);
```

Comment:
I'm not sure we should be checking `@data_representation` on non-topic types, since officially it's just for constraining QoS.

Reply:
I believe the most recent commit (4324ed7) should fix this for code that I added. There is at least 1 problem in gen_union that already existed and many in gen_struct that will need to be resolved.  I'm not sure that should be part of this PR considering how long it has been in review.